### PR TITLE
Add Cluster Setup documentation

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this?**
/kind documentation

**What does this PR do / why we need it**:
This PR adds the documentation for cluster setup.

**Which issue(s) this PR fixes**:

Fixes part of #4894 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 1]
```diff
@@ -1,8 +1,101 @@
...
+datadog-operator.v0.6.0             DatadogAgent, DatadogMetric, DatadogMonitor
+service-binding-operator.v0.9.1     ServiceBinding, ServiceBinding
+```
+If you do not see your installed operator in the list, follow the [troubleshooting guide](#troubleshoot-the-operator-installation) to find the issue and debug it.
+
+## Troubleshoot the Operator installation
+There are two ways to confirm that the operator has been installed properly.
+The examples you may see in this guide uses [Datadog Operator](https://operatorhub.io/operator/datadog-operator) and [Service Binding Operator](https://operatorhub.io/operator/service-binding-operator).
+1. Verify that its pod started and is in “Running” state.  
+  ```s****************hell
```
Comment (valaparthvi) [inline website/docs/getting-started/cluster-setup/kubernetes.md:70]: TODO: Fix this.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 2]
```diff
@@ -1,8 +1,101 @@
...
+This guide assumes that you have [installed minikube](https://minikube.sigs.k8s.io/docs/start/) on your system.
+
+If you are using a Kubernetes cluster other than minikube, this guide assumes that you have admin privileges to the cluster and are logged in with the admin user; operator installation is only possible with an admin user.
+
+**Agenda for this guide:**
+* [Install the OLM](#install-the-olm)
+* [Install the Service Binding Operator](#install-the-service-binding-operator)
+* [Install an operator](#install-an-operator)
+* [Verify the operator installation](#verify-the-operator-installation)
+
```
Comment (valaparthvi) [inline website/docs/getting-started/cluster-setup/kubernetes.md:18]: TODO: Add a note about enabling Ingress and Ingress-dns addon which is required for URL to work on minikube and perhaps kubernetes cluster.

**Review by dharmit — CHANGES_REQUESTED** [review]
Some changes based on a quick look.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 3]
```diff
@@ -1,8 +1,117 @@
...
+This guide assumes that you have [installed minikube](https://minikube.sigs.k8s.io/docs/start/) on your system.
+
+If you are using a Kubernetes cluster other than minikube, this guide assumes that you have admin privileges to the cluster and are logged in with the admin user; operator installation is only possible with an admin user.
+
+**Agenda for this guide:**
+* [Install the OLM](#install-the-olm)
+* [Install the Service Binding Operator](#install-the-service-binding-operator)
+* [Install an operator](#install-an-operator)
+* [Verify the operator installation](#verify-the-operator-installation)
+* [Enable Ingress addon](#enable-ingress-addon)
```
[the remark this answers is not in the document]
Comment (dharmit) [reply website/docs/getting-started/cluster-setup/kubernetes.md:18]: TBH, we don't need this Agenda because it shows up by default in the sidebar. And it's kinda sticky. Doesn't matter how much you scroll, it stays there. I would not put it separately in the doc.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 4]
```diff
@@ -1,8 +1,117 @@
...
+
+## Install the OLM
+The Operator Lifecycle Manager(OLM) is a component of the Operator Framework, an open source toolkit to manage Kubernetes native applications, called Operators, in a streamlined and scalable way.[(Source)](https://olm.operatorframework.io/)
```
[the remark this answers is not in the document]
Comment (valaparthvi) [reply website/docs/getting-started/cluster-setup/kubernetes.md:21]: Shouldn't we put this in the Operator doc?

[the remark this answers is not in the document]
Comment (dharmit) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: Yes we should. But at the moment, we are adding this doc. So we should put it here. If operator doc was ready, we could have linked to it. Once we have it ready, we could replace this paragraph with a link to it.

**Comment by dharmit** [comment]
@valaparthvi Update Changelog.md

File: website/docs/getting-started/cluster-setup/kubernetes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kadel) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: Rather than coming up with our own definition of what Operator is we should link to existing Kubernetes documentation 

https://kubernetes.io/docs/concepts/extend-kubernetes/operator/

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 5]
```diff
@@ -1,8 +1,111 @@
...
+```shell
+$ odo catalog list services
+Services available through Operators
+NAME                                CRDs
+datadog-operator.v0.6.0             DatadogAgent, DatadogMetric, DatadogMonitor
+service-binding-operator.v0.9.1     ServiceBinding, ServiceBinding
+```
+If you do not see your installed operator in the list, follow the [troubleshooting guide](#troubleshoot-the-operator-installation) to find the issue and debug it.
+
+## Enable Ingress addon
```
Comment (kadel) [inline website/docs/getting-started/cluster-setup/kubernetes.md:61]: I would move this section above "Install Operators". Enabling ingress is more important, as without it users won't be even able to access their applications.

File: website/docs/getting-started/cluster-setup/kubernetes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dharmit) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: We can and should surely link to k8s docs whenever it makes sense. But we should also make things simpler to understand for users who don't want/need to know things from k8s perspective.

Having said that, in this case, the `Motivatoin` section of the doc is pretty good and can be used to explain Operators instead of using what I suggested.

But overall, I hope that odo docs attempt to make k8s things simpler to understand for users. If that means coming up with our own definitions, so be it. Code should not be the only place where odo follows the philosophy of making k8s abstractions simple for users.

[the remark this answers is not in the document]
Comment (kadel) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: >  Code should not be the only place where odo follows the philosophy of making k8s abstractions simple for users.

agreed 

> If that means coming up with our own definitions,

That is dangerous. We should not come up with our own definitions for stuff that we did no created. We should no try to redefine things.

[the remark this answers is not in the document]
Comment (kadel) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: Why not use already existing definition from  https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ or https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator ? Do you think that those are too technical?

[the remark this answers is not in the document]
Comment (dharmit) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: > That is dangerous. 

The other way of looking at it is trying to do ELI5. Instead of really trying to do ELI5, we should try to explain things as if our readers have no knowledge of k8s.

> Why not use already existing definition from https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ or https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator ? Do you think that those are too technical?

I already said in my previous comment (https://github.com/openshift/odo/pull/4973#discussion_r686613855) that in this case, reusing what's in the "Motivation" section makes sense because it's written in really simple language.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 6]
```diff
@@ -1,8 +1,117 @@
 ---
 title: Kubernetes
-sidebar_position: 2
+sidebar_position: 1
 ---
 
 # Setup a minikube cluster
+Please note that this documentation is only useful in setting up a development environment, it is not recommended for a production environment.
```
[the remark this answers is not in the document]
Comment (valaparthvi) [reply website/docs/getting-started/cluster-setup/kubernetes.md]: About this, there's something that came up in my discussion with Marc. I'll try to phrase it here - 
If with odo, we are trying to reduce the gap between inner loop and outer loop, which is bringing the production environment closer to the developer or vice versa, do you think it makes sense to put this warning here? Wouldn't it ward off some users who are really trying to use odo with the intention of reducing the gap?

File: website/docs/getting-started/cluster-setup/kubernetes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (valaparthvi) [reply website/docs/getting-started/cluster-setup/kubernetes.md:15]: I will add the excerpt of Motivation from the official k8s doc, it's short and simple as compared to the RH article.

File: website/docs/getting-started/cluster-setup/kubernetes.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dharmit) [reply website/docs/getting-started/cluster-setup/kubernetes.md]: We are talking only about the k8s/ocp clusters that odo docs help spin up. We are not saying that odo is dev cluster only tool.

**Comment by valaparthvi** [comment]
@dharmit Can we get this one in first? Build for #4934 is broken atm because we removed some of the pages it is referring to, mainly cluster-setup pages. If this one gets in, 4934 will no longer break.

I am not happy with the current position of operator definition for OpenShift, but I couldn't think of any other place to put it, so suggestions are welcome.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
+---
+title: Kubernetes
+sidebar_position: 1
+---
+
+# Setup a Kubernetes cluster
+*Note that this guide is only helpful in setting up a development environment; this setup is not recommended for a production environment.*
+
+## Prerequisites
+* This guide assumes that you have a Kubernetes cluster setup, this could also be a [minikube](https://minikube.sigs.k8s.io/docs/start/) cluster.
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:10]: Because a minikube cluster **is** a Kubernetes cluster:

```suggestion
* This guide assumes that you have a Kubernetes cluster setup, this could for example be a [minikube](https://minikube.sigs.k8s.io/docs/start/) cluster.
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
+---
+title: Kubernetes
+sidebar_position: 1
+---
+
+# Setup a Kubernetes cluster
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:6]: We should be consistent in the form used for the titles: **Setup a Kubernetes cluster** vs **Setting up a Kubernetes cluster**. The second form seems more appropriate to me.

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+*Note that this guide is only helpful in setting up a development environment; this setup is not recommended for a production environment.*
+
+## Prerequisites
+* This guide assumes that you have a Kubernetes cluster setup, this could also be a [minikube](https://minikube.sigs.k8s.io/docs/start/) cluster.
+* This guide also assumes that you have admin privileges to the cluster, since the operator installation is only possible with an admin user.
+
+## Enable Ingress
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:13]: Perhaps **Enabling Ingress** ?

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+To access an application externally, odo creates a URL; installing an ingress-controller helps in using this feature on a Kubernetes cluster.
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:14]:
```suggestion
To access an application externally, you will create *URLs* using odo, which are implemented (materialized?) on a Kubernetes cluster by ingress resources; installing an ingress-controller helps in using this feature on a Kubernetes cluster.
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+To enable the Ingress feature on a Kubernetes cluster _other than minikube_, run the following command to install the default setup:
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v0.48.1/deploy/static/provider/cloud/deploy.yaml
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:18]: Please indicate that this command is specific to the `nginx` ingress controller. Also, you could point to the upstream doc for more details on installation: https://kubernetes.github.io/ingress-nginx/deploy/

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+>
+>People who run workloads on Kubernetes often like to use automation to take care of repeatable tasks. The Operator pattern captures how you can write code to automate a task beyond what Kubernetes itself provides.
+> [(Source)](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/#motivation)
+[//]: # (Move until here)
+
+To install an Operator, we will first need to install OLM [(Operator Lifecycle Manager)](https://olm.operatorframework.io/) on the cluster.
+```shell
+curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.3/install.sh | bash -s v0.18.3
+```
+Running the script will take some time to install all the necessary resources in the Kubernetes cluster including the `OperatorGroup` resource,....
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:43]: Please complete the resources list

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+## Installing the Service Binding Operator
+odo uses [Service Binding Operator](https://operatorhub.io/operator/service-binding-operator) to provide the `odo link` feature which helps to connect an odo component to a service or another component.
+
+Operators can be installed in a specific namespace or across the cluster(i.e. in all the namespaces).
+```shell
+kubectl create -f https://operatorhub.io/install/service-binding-operator.yaml
+```
+Running the command will create the necessary resource in the `operators` namespace.
+
+If you want to access this resource from other namespaces as well, add your target namespace to `.spec.targetNamespaces` list.
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:56]: Please specify on which file to add this

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+See [Verifying the Operator installation](#verifying-the-operator-installation) to ensure that the Operator was installed successfully.
+
+## Installing an Operator
+1. Visit the [OperatorHub](https://operatorhub.io) website.
+2. Search for an Operator of your choice.
+3. Navigate to its detail page.
+4. Click on **Install**.
+5. Follow the instruction in the installation popup. Please make sure to install the Operator in your desired namespace or cluster-wide, depending on your choice and the Operator capability.
+6. [Verifying the Operator installation](#verifying-the-operator-installation).
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:66]:
```suggestion
6. [Verify the Operator installation](#verifying-the-operator-installation).
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+```shell
+kubectl create -f https://operatorhub.io/install/service-binding-operator.yaml
+```
+Running the command will create the necessary resource in the `operators` namespace.
+
+If you want to access this resource from other namespaces as well, add your target namespace to `.spec.targetNamespaces` list.
+
+See [Verifying the Operator installation](#verifying-the-operator-installation) to ensure that the Operator was installed successfully.
+
+## Installing an Operator
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:60]: You can add a little sentence to introduce what will be done in the following steps, smth similar to:

```
To install an operator from the OperatorHub website:
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+Services available through Operators
+NAME                                CRDs
+datadog-operator.v0.6.0             DatadogAgent, DatadogMetric, DatadogMonitor
+service-binding-operator.v0.9.1     ServiceBinding, ServiceBinding
+```
+If you do not see your installed Operator in the list, follow the [troubleshooting guide](#troubleshoot-the-operator-installation) to find the issue and debug it.
+
+## Troubleshooting the Operator installation
+There are two ways to confirm that the Operator has been installed properly.
+The examples you may see in this guide uses [Datadog Operator](https://operatorhub.io/operator/datadog-operator) and [Service Binding Operator](https://operatorhub.io/operator/service-binding-operator).
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:87]:
```suggestion
The examples you may see in this guide use [Datadog Operator](https://operatorhub.io/operator/datadog-operator) and [Service Binding Operator](https://operatorhub.io/operator/service-binding-operator).
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+1. Verify that its pod started and is in “Running” state.
+  ```shell
+  kubectl get pods -n operators
+  ```
+The output can look similar to:
+  ```shell
+  $ kubectl get csv -n operators
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:94]:
```suggestion
  $ kubectl get pods -n operators
```

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+  NAME                                       READY   STATUS    RESTARTS   AGE
+  datadog-operator-manager-5db67c7f4-hgb59   1/1     Running   0          2m13s
+  service-binding-operator-c8d7587b8-lxztx   1/1     Running   5          6d23h
+  ```
+2. Verify that the csv is in Succeeded or Installing phase.
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:99]:
```suggestion
2. Verify that the ClusterServiceVersion (csv) resource is in Succeeded or Installing phase.
```

File: website/docs/getting-started/cluster-setup/openshift.md [hunk 19]
```diff
@@ -0,0 +1,58 @@
+---
+title: OpenShift
+sidebar_position: 2
+---
+
+# Setup an OpenShift cluster
+*Note that this guide is only helpful in setting up a development environment; this setup is not recommended for a production environment.*
+
+## Prerequisites
+* This guide assumes that you have an OpenShift cluster setup, this could also be a [crc](https://crc.dev/crc/#installing-codeready-containers_gsg) cluster.
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/openshift.md:10]:
```suggestion
* This guide assumes that you have an OpenShift cluster setup, this could for example be a [crc](https://crc.dev/crc/#installing-codeready-containers_gsg) cluster.
```

(crc being an OpenShift cluster)

File: website/docs/getting-started/cluster-setup/openshift.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+**What are Operators?**
+>The Operator pattern aims to capture the key aim of a human operator who is managing a service or set of services. Human operators who look after specific applications and services have deep knowledge of how the system ought to behave, how to deploy it, and how to react if there are problems.
+>
+>People who run workloads on Kubernetes often like to use automation to take care of repeatable tasks. The Operator pattern captures how you can write code to automate a task beyond what Kubernetes itself provides.
+> [(Source)](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/#motivation)
+[//]: # (Move until here)
+
+## Install the Service Binding Operator
+odo uses [Service Binding Operator](https://operatorhub.io/operator/service-binding-operator) to provide the `odo link` feature which helps connect an odo component to a service or another component.
+
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/openshift.md:22]: Please introduce what we will do with the steps below, for example:

```
To install the Service binding operator from the OpenShift web console:
```

File: website/docs/getting-started/cluster-setup/openshift.md [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+* This guide also assumes that you have admin privileges to the cluster, since Operator installation is only possible with an admin user.
+[//]: # (Move this section to Architecture > Service Binding or create a new Operators doc)
+**What are Operators?**
+>The Operator pattern aims to capture the key aim of a human operator who is managing a service or set of services. Human operators who look after specific applications and services have deep knowledge of how the system ought to behave, how to deploy it, and how to react if there are problems.
+>
+>People who run workloads on Kubernetes often like to use automation to take care of repeatable tasks. The Operator pattern captures how you can write code to automate a task beyond what Kubernetes itself provides.
+> [(Source)](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/#motivation)
+[//]: # (Move until here)
+
+## Install the Service Binding Operator
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/openshift.md:20]:
```suggestion
## Installing the Service Binding Operator
```

Also in other places

File: website/docs/getting-started/cluster-setup/openshift.md [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+3. Search for _**Service Binding Operator**_ in the search box under **All Items**.
+4. Click on the **Service Binding Operator**; this should open a side pane.
+5. Click on the **Install** button on the side pane; this should open an **Install Operator** page.
+6. Make sure the **Installation mode** is set to "_All namespaces on the cluster(default)_"; **Installed Namespace** is set to "_openshift-operators_"; and **Approval Strategy** is "_Automatic_".
+7. Click on the **Install** button.
+8. Wait until the Operator is installed.
+9. Once the Operator is installed, you should see **_Installed operator - ready for use_**, and a **View Operator** button appears on the page.
+10. Click on the **View Operator** button; this should take you to Operators > Installed Operators > Operator details page, and you should be able to see details of your Operator.
+
+## Install an Operator
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/openshift.md:34]: Please introduce what will be done

File: website/docs/getting-started/cluster-setup/kubernetes.md (hunk 12 above)
Comment (valaparthvi) [reply website/docs/getting-started/cluster-setup/kubernetes.md:43]: This seems like something specific to OLM, do you think it is required?

Comment (feloy) [reply website/docs/getting-started/cluster-setup/kubernetes.md:43]: No, so probably just remove the 3 dots

File: website/docs/getting-started/cluster-setup/kubernetes.md [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+## Enabling Ingress
+To access an application externally, you will create _URLs_ using odo, which are implemented on a Kubernetes cluster by Ingress resources; installing an Ingress controller helps in using this feature on a Kubernetes cluster.
+
+**Minikube:** To install an Ingress controller on a minikube cluster, enable the **ingress** addon with the following command:
+```shell
+minikube addons enable ingress
+````
+To learn more about ingress addon, see [the documentation on Kubernetes website](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/).
+
+*Other Kubernetes Cluster**: To enable the Ingress feature on a Kubernetes cluster _other than minikube_, using the NGINX Ingress controller see [the official NGINX Ingress controller installation documentation](https://kubernetes.github.io/ingress-nginx/deploy/).
```
Comment (feloy) [inline website/docs/getting-started/cluster-setup/kubernetes.md:22]: It is not clear what covers the **Other Kubernetes cluster**. It could be better to use a section:

```suggestion
### Other Kubernetes Cluster

To enable the Ingress feature on a Kubernetes cluster _other than minikube_, using the NGINX Ingress controller see [the official NGINX Ingress controller installation documentation](https://kubernetes.github.io/ingress-nginx/deploy/).
```

and idem for **minikube**



If you prefer to keep this presentation, please just add a missing \*

**Review by dharmit — APPROVED** [review]
